File: packages/push/ios/PushErrorCode.swift
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

/// Stable error code strings passed to `rejecter(code, message, nil)`.
///
/// These values form the contract with the TypeScript `PushErrorCode` union in
/// `@ping-identity/rn-push`. Keep in sync with `PushErrorMapper` (Android) and
/// the `pushErrorCode(_:)` function in `RNPingPushCommon`.
enum PushErrorCode: String {
  case notInitialized = "not_initialized"
  case initializationFailed = "initialization_failed"
  case invalidUri = "invalid_uri"
  case missingRequiredParameter = "missing_required_parameter"
  case invalidParameterValue = "invalid_parameter_value"
  case invalidPushType = "invalid_push_type"
  case invalidPlatform = "invalid_platform"
  case storageFailure = "storage_failure"
  case deviceTokenNotSet = "device_token_not_set"
  case noHandlerForPlatform = "no_handler_for_platform"
  case messageParsingFailed = "message_parsing_failed"
  case credentialNotFound = "credential_not_found"
  case credentialLocked = "credential_locked"
  case duplicateCredential = "duplicate_credential"
  case notificationNotFound = "notification_not_found"
  case policyViolation = "policy_violation"
  case registrationFailed = "registration_failed"
  case networkFailure = "network_failure"
}

File: packages/push/android/src/main/java/com/pingidentity/rnpush/PushErrorCode.kt
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */
package com.pingidentity.rnpush

/**
 * Stable error code strings passed to `promise.reject(code, message)`.
 *
 * These values form the contract with the TypeScript `PushErrorCode` union in
 * `@ping-identity/rn-push`. Keep in sync with [PushErrorMapper] and the iOS
 * `PushErrorCode` Swift enum.
 */
internal object PushErrorCode {
    const val NOT_INITIALIZED = "not_initialized"
    const val INITIALIZATION_FAILED = "initialization_failed"
    const val INVALID_URI = "invalid_uri"
    const val MISSING_REQUIRED_PARAMETER = "missing_required_parameter"
    const val INVALID_PARAMETER_VALUE = "invalid_parameter_value"
    const val INVALID_PUSH_TYPE = "invalid_push_type"
    const val INVALID_PLATFORM = "invalid_platform"
    const val STORAGE_FAILURE = "storage_failure"
    const val DEVICE_TOKEN_NOT_SET = "device_token_not_set"
    const val NO_HANDLER_FOR_PLATFORM = "no_handler_for_platform"
    const val MESSAGE_PARSING_FAILED = "message_parsing_failed"
    const val CREDENTIAL_NOT_FOUND = "credential_not_found"
    const val CREDENTIAL_LOCKED = "credential_locked"
    const val DUPLICATE_CREDENTIAL = "duplicate_credential"
    const val NOTIFICATION_NOT_FOUND = "notification_not_found"
    const val POLICY_VIOLATION = "policy_violation"
    const val REGISTRATION_FAILED = "registration_failed"
    const val NETWORK_FAILURE = "network_failure"
}
